src/include/gurt/debug.h

```diff
@@ -279,8 +280,10 @@ int d_register_alt_assert(void (*alt_assert)(const int, const char*,
 
 #define D_ASSERT(e)							\
 	do {								\
-		if (!(e))						\
-			D_FATAL("Assertion '%s' failed\n", #e);	\
+		if (!(e)) {						\
+			D_FATAL("Assertion '%s' failed\n", #e);		\
+			d_log_sync();					\
```

Comment:
what about sigsegv or other such signals?   I also would want the log dumped in such cases